app/src/main/java/co/icanteach/projectx/domain/FetchPopularTvShowUseCase.kt

```diff
@@ -12,14 +14,38 @@ class FetchPopularTvShowUseCase @Inject constructor(
 ) {
 
     fun fetchMovies(page: Int): Observable<Resource<List<PopularTvShowItem>>> {
+        return Observable.concatArray(fetchMoviesFromLocal(), fetchMoviesFromRemote(page))
+    }
+
+    private fun fetchMoviesFromRemote(page: Int): Observable<Resource<List<PopularTvShowItem>>> {
+        return repository
+            .fetchMoviesFromRemote(page)
+            .map { resource ->
+                Resource(
+                    status = resource.status,
+                    data = resource.data?.let { mapper.mapFromResponse(it) },
+                    error = resource.error
+                )
+            }
+            .doOnNext {
```

Comment:
what do you think if we can provide extension function for this kind of conditions doOnLoading, doOnSuccess, doOnError etc.

Reply:
I couldn't a good solution for this. what do you prefer for it? When I tried to store movies in rxMap method, I have to manege Stream state  because of loading State extension in consume method and I don't want to another condition control

Reply:
I created doOnSuccess extension method for store movies